src/Schema/Directives/Traits/RelationDirectiveHelpers.php

```diff
@@ -25,20 +24,14 @@ protected function relation(): string
         return $this->directiveArgValue('relation', $this->nodeName());
     }
 
-    /**
-     * @param  array<string, mixed>  $args
-     *
-     * @return \Closure(QueryBuilder|\Illuminate\Database\Eloquent\Builder<\Illuminate\Database\Eloquent\Model>|\Illuminate\Database\Eloquent\Relations\Relation<\Illuminate\Database\Eloquent\Model>): void
-     */
+    /** @param  array<string, mixed>  $args */
     protected function makeBuilderDecorator(mixed $root, array $args, GraphQLContext $context, ResolveInfo $resolveInfo): \Closure
     {
-        return function (object $builder) use ($root, $args, $context, $resolveInfo): void {
+        return function (Builder $builder) use ($root, $args, $context, $resolveInfo): void {
             if ($builder instanceof Relation) {
                 $builder = $builder->getQuery();
             }
```

Comment:
Is that even necessary anymore, now that `Relation` implements the interface?

Reply:
The interface is empty. It doesn't change anything.

Technically, `getModel` works on `EloquentBuilder` and `Relation`, but not on `QueryBuilder`.

But phpstan will throw the error `Call to an undefined method Illuminate\Database\Eloquent\Relations\Relation::getModel()`, because it does not account for the `ForwardsCalls` trait in `Relation`.

So for the time being, I would keep the check.